EvidenceApi.Tests/V1/Gateways/EvidenceGatewayTests.cs

```diff
@@ -857,6 +857,36 @@ public void UpdateResidentIdForDocumentSubmissionUpdatesDocumentSubmissionsCorre
             result.Total.Should().Be(3);
 
         }
+        [Test]
+        public void UpdateVisibilityForDocumentSubmissionUpdatesDocumentSubmissionsCorrectly()
+        {
+            var resident = TestDataHelper.Resident();
+            var team = "Fake team";
+            DatabaseContext.Residents.Add(resident);
+            DatabaseContext.SaveChanges();
+
+            var evidenceRequest = TestDataHelper.EvidenceRequest();
+            var documentsubmission1 = TestDataHelper.DocumentSubmissionWithResidentId(resident.Id, evidenceRequest);
+            documentsubmission1.Team = team;
+            var documentsubmission2 = TestDataHelper.DocumentSubmissionWithResidentId(resident.Id, evidenceRequest);
+            documentsubmission2.Team = team;
+            var documentsubmission3 = TestDataHelper.DocumentSubmissionWithResidentId(resident.Id, evidenceRequest);
+            documentsubmission3.Team = team;
+
+            DatabaseContext.EvidenceRequests.Add(evidenceRequest);
+            DatabaseContext.DocumentSubmissions.Add(documentsubmission1);
+            DatabaseContext.DocumentSubmissions.Add(documentsubmission2);
+            DatabaseContext.DocumentSubmissions.Add(documentsubmission3);
+
+            DatabaseContext.SaveChanges();
+
+            _classUnderTest.UpdateVisibilityDocumentSubmission(documentsubmission2.Id, true);
+
+            var result = _classUnderTest.GetPaginatedDocumentSubmissionsByResidentId(resident.Id, team);
+            result.Total.Should().Be(2);
+            result.DocumentSubmissions[0].IsHidden.Should().BeFalse();
```

Comment:
Could perhaps test here using linq that all returned submissions are visible. Also would be good to test that the correct submission id is missing from the results, so it doesn't rely on count alone.

Reply:
Addressed (I think) in [2a101cd](https://github.com/LBHackney-IT/evidence-api/pull/235/commits/2a101cd7e3b56cd72918edc5935aec5ca4a09c28)

Reply:
That ensures the correct records are returned. You could also do `result.DocumentSubmissions.All(x => x.IsHidden).Should().BeFalse();` rather than using the index, but this will work too.